### PR TITLE
Chats

### PR DESCRIPTION
**Pull request by moezAhmad**
# Pull Request Template

## Description

**Provide a description of the changes you made in this PR**

- Add Github issue link:
  Make sure to use closing keywords

## Reviewers

Please request a review for this PR.

> Make sure to link an issue with this PR using closing keywords in the description or from the development tab

### CODE REVIEW

File: src/types/app/(private)/chat/index.ts
Comment (Waleed-abdullah) [inline src/types/app/(private)/chat/index.ts:1]: no need to add directories for teh entire path just create one meaningful subdirectory like types/chat/index.ts

**Review by Waleed-abdullah — CHANGES_REQUESTED** [review]

File: src/pages/api/message/send.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { sendErrorResponse } from '../../../utils/errorHandler';
```
Comment (Waleed-abdullah) [inline src/pages/api/message/send.ts]: updat import to use @/utils

File: src/pages/api/message/request.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { sendErrorResponse } from '../../../utils/errorHandler';
```
Comment (Waleed-abdullah) [inline src/pages/api/message/request.ts]: same import here

File: src/pages/api/message/deny.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { sendErrorResponse } from '../../../utils/errorHandler';
```
Comment (Waleed-abdullah) [inline src/pages/api/message/deny.ts]: same import here

File: src/pages/api/message/accept.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { sendErrorResponse } from '../../../utils/errorHandler';
```
Comment (Waleed-abdullah) [inline src/pages/api/message/accept.ts]: same import here

File: src/pages/api/message/accept.ts
Comment (Waleed-abdullah) [inline src/pages/api/message/accept.ts:1]: prefer using app router api routes, they are exposed using route.ts file

File: src/components/MessageRequests/index.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  const acceptMessageRequest = async (sender_id: string) => {
+    await axios.post('/api/message/accept', { sender_id });
+    setMessageRequests((prev) =>
+      prev.filter((request) => request.user_id !== sender_id),
+    );
+    router.refresh();
+  };
+
+  const denyMessageRequest = async (sender_id: string) => {
+    await axios.post('/api/message/deny', { sender_id });
```
Comment (Waleed-abdullah) [inline src/components/MessageRequests/index.tsx:47]: does this rerender properly

File: src/app/warrior/trainer-marketplace/types/pusher.d.ts
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/types/pusher.d.ts:1]: move file to types/pusher/index.ts instead

File: src/app/warrior/trainer-marketplace/components/TrainerCard.tsx [hunk 6]
```diff
@@ -25,12 +23,12 @@ const TrainerCard: React.FC<Trainer> = ({
...
+      {/* <Image
         src={users.profile_pic || ''}
         alt={users.name}
         width={200}
         height={200}
-      />
+      /> */}
       <div className='grow flex flex-col flex-start gap-2 ml-6'>
         <p className='font-bold'>{users.name}</p>
         <div className='flex flex-row flex-start gap-2'>
```
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/components/TrainerCard.tsx]: shouldnt we show trainer iage here

File: src/app/(private)/chat/[chat_id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import { authOptions } from '@/pages/api/auth/[...nextauth]';
+import prisma from '@/lib/database/prisma';
+import { getServerSession } from 'next-auth';
+import { notFound } from 'next/navigation';
+import { FC } from 'react';
+import Messages from '../../../../components/Messages';
```
Comment (Waleed-abdullah) [inline src/app/(private)/chat/[chat_id]/page.tsx]: update import to @/components/messages

**Review by Waleed-abdullah — CHANGES_REQUESTED** [review]

File: src/components/MessageRequests/index.tsx (hunk 5 above)
Comment (moezAhmad) [reply src/components/MessageRequests/index.tsx:47]: yes, new messages are displayed

File: src/app/warrior/trainer-marketplace/components/TrainerCard.tsx (hunk 6 above)
Comment (moezAhmad) [reply src/app/warrior/trainer-marketplace/components/TrainerCard.tsx]: Updated

File: src/app/warrior/trainer-marketplace/[user_id]/page.tsx [hunk 8]
```diff
@@ -2,7 +2,7 @@
 import React, { useState, useEffect } from 'react';
 import { fetchTrainer } from '../services/trainerService';
 import { useParams } from 'next/navigation';
-import { DetailedTrainer } from '../types/';
+import { DetailedTrainer } from '../../../../types/marketplace';
```
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/[user_id]/page.tsx]: again need @/type/ import, always use absolute imports when going more than one level down or up in the folder tree

File: src/app/warrior/trainer-marketplace/components/TrainerCard.tsx [hunk 9]
```diff
@@ -5,7 +5,7 @@ import Image from 'next/image';
 import { useRouter } from 'next/navigation';
 import Location from '@/assets/trainer-marketplace/location.svg';
 import Experience from '@/assets/trainer-marketplace/experience.svg';
-import { Trainer } from '../types';
+import { Trainer } from '../../../../types/marketplace';
```
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/components/TrainerCard.tsx]: same here

File: src/app/warrior/trainer-marketplace/components/TrainerList.tsx [hunk 10]
```diff
@@ -3,7 +3,7 @@ import React, { useState, useEffect } from 'react';
 import TrainerCard from './TrainerCard';
 import { fetchTrainers } from '../services/trainerService';
 import { useTrainerFilter } from '../context/TrainerFilterContext';
-import { Trainer } from '../types';
+import { Trainer } from '../../../../types/marketplace';
```
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/components/TrainerList.tsx]: same here, I assume vscode automatically updated these when you moved the file and you missed them

File: src/app/warrior/trainer-marketplace/context/TrainerFilterContext.tsx [hunk 11]
```diff
@@ -1,5 +1,5 @@
 import React, { createContext, useContext, useState } from 'react';
-import { TrainerFilters } from '../types';
+import { TrainerFilters } from '../../../../types/marketplace';
```
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/context/TrainerFilterContext.tsx]: same

File: src/app/warrior/trainer-marketplace/services/trainerService.ts [hunk 12]
```diff
@@ -1,4 +1,4 @@
-import { TrainerFilters } from '../types';
+import { TrainerFilters } from '../../../../types/marketplace';
```
Comment (Waleed-abdullah) [inline src/app/warrior/trainer-marketplace/services/trainerService.ts]:
```suggestion
import { TrainerFilters } from '@/types/marketplace';
```

**Review by Waleed-abdullah — APPROVED** [review]
Restis good just update the requested changes and mege once done

**Pull request merged**